user_modules/sequences/actions.lua

```diff
@@ -137,15 +137,19 @@ end
 --! @parameters: none
 --! @return: none
 --]]
-local function allowSDL(self)
-  self.hmiConnection:SendNotification("SDL.OnAllowSDLFunctionality", {
+local function allowSDL()
+  local event = events.Event()
+  event.matches = function(e1, e2) return e1 == e2 end
+  m.getHMIConnection():SendNotification("SDL.OnAllowSDLFunctionality", {
     allowed = true,
     source = "GUI",
     device = {
       id = utils.getDeviceMAC(),
       name = utils.getDeviceName()
     }
   })
+  RUN_AFTER(function() m.getHMIConnection():RaiseEvent(event, "Allow SDL event") end, 500)
```

Comment:
Is 500ms necessary? I could see that causing a lot of overhead, given that this is part of every test.

Reply:
@jacobkeeler Unfortunately this timeout is required for SDL in order it has enough time to recalculate device permissions. (applicable for SDL in `EXTERNAL_PROPRIETARY` mode). We could try to reduce this number to a lower value (e.g. for 100-200ms)

Reply:
@jacobkeeler We've just tried to run the tests with 250ms timeout and aprox. 10% of runs failed. So I propose to leave 500ms to have more stable runs.
Taking into account that average simple test script is executed during 10 sec. we will have only 5% increase with 500ms delay.